osu.Game/Screens/Play/PlayerLoader.cs

```diff
@@ -377,6 +385,11 @@ protected virtual void OnPlayerLoaded()
 
         private void restartRequested()
         {
+            if (CurrentPlayer != null)
+            {
+                isHotKeyRestart = CurrentPlayer.Configuration.AutomaticallySkipIntro;
+            }
```

Comment:
I'm so confused.

What is this doing?

Reply:
On each restart a new player gets created along side with a new player configuration. If the intro should be skipped, we need to store this or else it will be overwritten (by a new player config). About the null check: I experienced a few crashes because the current player was sometimes null.